src/Assert.php

```diff
@@ -1822,11 +1822,24 @@ public static function countBetween($array, $min, $max, $message = '')
      */
     public static function isList($array, $message = '')
     {
-        if (!\is_array($array) || $array !== \array_values($array)) {
+        if (!\is_array($array)) {
             static::reportInvalidArgument(
                 $message ?: 'Expected list - non-associative array.'
             );
         }
+
+        if ($array === \array_values($array)) {
+            return;
+        }
```

Comment:
This is a double check with an extra copy of the array. You can check both cases with a single pass through the array. This is more efficient.

Reply:
@eugene-borovov how?

Reply:
@boesing checked this in the symfony polyfill repo (https://github.com/symfony/polyfill/pull/356)

https://3v4l.org/ruYVL

This check is more performant than the lower one, since this is the happy (usual) path, the early return should give better perormance than not doing this check most of the time

Reply:
Thank you for the clarification. I thought more about the failed path.